docs/contributing/supporting_new_profiler.md

```diff
@@ -0,0 +1,231 @@
+of the many `pandas.read_*` functions with specific arguments tailored to reading your
+particular file format. Whatever you do, you need to ensure that the output returned by
+calling the `read` class method follows the schema that you have defined before. Again,
+you can look at existing code for inspiration.
+
+Don't forget to test that your reader creates expected output by adding a few test cases
+into a module that you place in `tests/unit/infrastructure/application/`. To test your
+reader you probably want to add some appropriate files into `tests/data/<tool>`, too, and
+make them accessible with a pytest fixture function which you place into `tests/conftest.py`.
+Take a look at the kraken2 fixture:
+
+```python
+@pytest.fixture(scope="module")
+def kraken2_data_dir(data_dir: Path) -> Path:
+    """Provide the path to the kraken2 data directory."""
+    return data_dir / "kraken2"
+```
+
+### 3. Validation & Transformation
+
+ The
+validation is automatically performed by decorating with [`pandera.check_types`](https://pandera.readthedocs.io/en/stable/schema_models.html) and annotating the
+`transform` class method of the standardisation service, for example,
+
+```python
+import pandera as pa
+from pandera.typing import DataFrame
+
+from taxpasta.application import ProfileStandardisationService
+from taxpasta.domain import StandardProfile
+
+from .kraken2_profile import Kraken2Profile
+
+
+class Kraken2ProfileStandardisationService(ProfileStandardisationService):
+    @classmethod
+    @pa.check_types
+    def transform(
+        cls, profile: DataFrame[Kraken2Profile]
+    ) -> DataFrame[StandardProfile]:
+```
+
+Finally, we need to transform the specific taxonomic profile into our standard
+profile. Similarly to the profile reader, there exists an abstract
+[`ProfileStandardisationService`][taxpasta.application.ProfileStandardisationService]
+that you need to inherit from. The new module should be placed into
+`taxpasta/infrastructure/application/` and the naming should follow the
+conventions, as an example, `Kraken2ProfileStandardisationService` class in a
+`kraken2_profile_standardisation_service.py` module.
+
+```python
+import pandera as pa
+from pandera.typing import DataFrame
+
+from taxpasta.application import ProfileStandardisationService
+from taxpasta.domain import StandardProfile
+
+from .kraken2_profile import Kraken2Profile
+
+
+class Kraken2ProfileStandardisationService(ProfileStandardisationService):
+    @classmethod
+    @pa.check_types(lazy=True)
+    def transform(
+        cls, profile: DataFrame[Kraken2Profile]
+    ) -> DataFrame[StandardProfile]:
+```
+
+The `pa.check_types` decorator validates the class method's input and output
+using the type annotations and the defined schema models.
+
+The `transform` class method itself needs to modify the given `pandas.DataFrame` such that
+the returned result looks like the [`StandardProfile`][].
```

Comment:
Missing link? Expected for now?

Reply:
Ooops, missed that now. I'll fix it in a separate commit.